data-prepper-plugins/geoip-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/extension/MaxMindConfig.java

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright OpenSearch Contributors
+ *  PDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.plugins.processor.extension;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import jakarta.validation.constraints.AssertTrue;
+import jakarta.validation.constraints.Min;
+
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+
+public class MaxMindConfig {
+    //TODO: Add validations to database paths
+    //TODO: Make default path to be a public CDN endpoint
+    private static final List<String> DEFAULT_DATABASE_PATHS = new ArrayList<>();
+    private static final Duration DEFAULT_DATABASE_REFRESH_INTERVAL = Duration.ofDays(7);
+    private static final int DEFAULT_CACHE_SIZE = 4096;
+
+    @JsonProperty("database_paths")
+    private List<String> databasePaths = DEFAULT_DATABASE_PATHS;
+    @JsonProperty("database_refresh_interval")
+    private Duration databaseRefreshInterval = DEFAULT_DATABASE_REFRESH_INTERVAL;
+    @JsonProperty("cache_size")
+    @Min(1)
+    //TODO:  Add a Max limit on cache size
```

Comment:
Do you have any thoughts on the maximum cache size? How big is each entry?

Reply:
I don't have any right now, but it's something to think about as it can abused if max limit is not added